### DISCUSSION TITLE
Leftover Ghosted Objects

### DISCUSSION THREAD

**Discussion by JoopBman** [anyproto/contributors#266] [Any-Block & Anytype-Heart]
Whenever i create a project i always nest things inside it. I love the versatility of this. However; when a project is done, i either archive (tag) it or send it to bin/delete. Then i find that all nested objects are not going with them. They stay in the "active" places. From what i read it seems this is because of not accidentally removing objects you also use elsewhere.

This results in a **ghosttown of leftover objects** that become confusing or having to remember manually fix every nested level. The whole thing becomes poorly organized. I can't code (yet), but below might solve the issue, while still preventing accidental deletions.

1.  When deleting/binning; **check if the nested objects are also nested somewhere else. If not delete bin them also**. If anyone want to keep certain objects that exist without nesting elsewhere that should be saved, the could nest them under a "safe" so they are always nested. Alternatively there could be a "pin" function that prevents deletion/binning that object.

2. **There should be an archive function**, just like deleting/binning. Archiving should nest the object (and all objects withing, just like with above) to an object that can be chosen by the user. For instance a collection. When hitting the Archive button without an Archive target set, it could ask to choose or create one.

**Comment by filip-anytype** [comment]
[Here's the relevant feature request from our forum.](https://community.anytype.io/t/nested-files-objects-deletion/2715)
